Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -480,12 +480,12 @@ AccessibilityRole AccessibilityNodeObject::determineAccessibilityRoleFromNode(Tr
     }
 
     // http://webkit.org/b/190138 Footers should become contentInfo's if scoped to body (and consequently become a landmark).
-    // It should remain a footer if scoped to main, sectioning elements (article, aside, nav, section) or root sectioning element (blockquote, details, dialog, fieldset, figure, td).
+    // It should remain a footer if scoped to main, sectioning elements (article, aside, nav, section).
     // https://w3c.github.io/html-aam/#el-footer
     if (node->hasTagName(footerTag)) {
-        if (!isDescendantOfElementType({ articleTag, asideTag, navTag, sectionTag, mainTag, blockquoteTag, detailsTag, dialogTag, fieldsetTag, figureTag, tdTag }))
+        if (!isDescendantOfElementType({ articleTag, asideTag, mainTag, navTag, sectionTag }))
```

Comment:
This removes the check for "root sectioning elements", which was added here:

https://github.com/WebKit/WebKit/commit/5ef5269eebb1ed66acf27ea3c622cee1661f34c0

I guess that makes sense since I can't seem to find this notion of root sectioning elements anywhere in a spec, but cc @cookiecrook just in case.